eng/common/scripts/SemVer.ps1

```diff
@@ -1,11 +1,19 @@
 <#
 .DESCRIPTION
-Parses a semver version string into its components and supports operations around it.
+Parses a semver version string into its components and supports operations around it that we use for versioning our packages.
+
+By default enforces prerelease label semantics:
+     If a prerelease label exists, it must be 'preview', and similar semantics used in our release guidelines
+
+See https://azure.github.io/azure-sdk/policies_releases.html#package-versioning
+
 Example: 1.2.3-preview.4
 Components: Major.Minor.Patch-PrereleaseLabel.PrereleaseNumber
+
+Note: A builtin Powershell version of SemVer exists in 'System.Management.Automation'. At this time, it does not parsing of PrereleaseNumber. It's name is also type accelerated to 'SemVer'.
 #>
 
-class SemVer {
+class AzureEngSemanticVersion {
```

Comment:
Honestly I'm not even sure we need to even define a class and as we can just use anonymous types (aka PSCustomObject under the hood), which is what I do in most of my scripts.

Reply:
Personally I prefer the strong types in general from a validation perspective, and makes adding functions with common names easy without worrying about naming collisions (admittedly the name collision on the class itself is annoying here). If we're going through the effort of creating the PSCustomObject, it's only one more step to getting to a full class, especially as the number of helper functions grows.